src/main.rs

```diff
@@ -16,29 +16,56 @@ use tokio_core::reactor::Core;
 use name_gen::random_msg;
 
 fn main() {
-    let token = get_token(); 
-
-    let epic = random_msg();
-    let tweet = DraftTweet::new(&epic);
-    
-    let mut core = Core::new().unwrap();
-    let handle = core.handle();
-    core.run(tweet.send(&token, &handle)).unwrap();
+    let token = get_token();
+
+    if token.is_err() {
+        println!("Failed to find a .env file: {:?}", token);
+    } else {
+        let epic = random_msg();
+        let tweet = DraftTweet::new(&epic);
+        
+        let mut core = Core::new().unwrap();
+        let handle = core.handle();
+        let result = core.run(tweet.send(&token.unwrap(), &handle));
+        match result {
+            Ok(_) => println!("Successfully posted tweet: {}", epic),
+            Err(err) => println!("Couldn't post tweet, reason: {}", err),
+        }
+    }
 }
 
-fn get_token() -> egg_mode::Token {
+fn get_token() -> Result<egg_mode::Token, TokenError> {
     dotenv().ok();
 
-    let consumer_key = env::var("CONSUMER_KEY").unwrap();
-    let consumer_secret = env::var("CONSUMER_SECRET").unwrap();
-    let access_token = env::var("ACCESS_TOKEN").unwrap();
-    let access_token_secret = env::var("ACCESS_TOKEN_SECRET").unwrap();
+    let consumer_key = env::var("CONSUMER_KEY");
```

Comment:
Using the enum definition of the `TokenError` type, we can write this as
```rust
let consumer_key = env::var("CONSUMER_KEY")
    .map_err(TokenError::ConsumerKey)?;
```
`map_err()` wraps the error if it has an error, and `?` will early-return any error case, and unwrap success cases.